lib/core/heartbeat.rb

```diff
@@ -21,6 +21,8 @@ def send(connector_settings)
         if connector_settings.connector_status_allows_sync?
           connector_instance = Connectors::REGISTRY.connector(connector_settings.service_type, connector_settings.configuration)
           doc[:status] = connector_instance.is_healthy? ? Connectors::ConnectorStatus::CONNECTED : Connectors::ConnectorStatus::ERROR
+          message = "Health check for 3d party service failed for connector [#{connector_settings.id}], service type [#{connector_settings.service_type}]. Check the application logs for more information."
+          doc[:error] = doc[:status] == Connectors::ConnectorStatus::ERROR ? message : nil
```

Comment:
```suggestion
          if connector_instance.is_healthy?
  doc[:status] = Connectors::ConnectorStatus::CONNECTED
  doc[:error] = nil
else
  doc[:status] = Connectors::ConnectorStatus::ERROR
  doc[:error] = "Health check for 3d party service failed for connector [#{connector_settings.id}], service type [#{connector_settings.service_type}]. Check the application logs for more information."
end
```

Reply:
Why would we replace 3 lines with 6?

Reply:
I proposed the change because I see you checked `connector_instance.is_healthy?` multiple times, directly or indirectly.

But I have no strong objection to keep it.